src/java.base/share/classes/jdk/crac/Core.java

```diff
@@ -140,18 +147,49 @@ private static void checkpointRestore1() throws
             }
         }
 
+        RestoreException restoreException = null;
         try {
             globalContext.afterRestore(null);
         } catch (RestoreException re) {
             if (checkpointException == null) {
-                throw re;
+                restoreException = re;
+            } else {
+                for (Throwable t : re.getSuppressed()) {
+                    checkpointException.addSuppressed(t);
+                }
             }
-            for (Throwable t : re.getSuppressed()) {
-                checkpointException.addSuppressed(t);
+        }
+
+        if (newArguments != null && newArguments.length() > 0) {
+            String[] args = newArguments.split(" ");
+            if (args.length > 0) {
+                try {
+                    Class<?> newMainClass = Class.forName(args[0], false,
+                        ClassLoader.getSystemClassLoader());
+                    Method newMain = newMainClass.getDeclaredMethod("main",
+                        String[].class);
+                    newMain.setAccessible(true);
```

Comment:
Good catch on adding the `setAccessible` call.  

Given that the VM is doing the lookup based on the command line parameters, we should be wrapping this with a `AccessController::doPrivileged` block as a SecurityManager shouldn't be able to block this accessibility request.

Reply:
I was unsure about SecurityManager, having it will be removed soon. However, while it's still there, and adding necessary code is not hard, it's worth fixing.

I'm trying to push the fix, but GitHub sends me "Internal Server Error". Hope this will be resolved soon.